src/index.ts

```diff
@@ -23,6 +23,8 @@ export * from './types';
 
 export { ConversationStore, MemoryStore } from './conversation-store';
 
+export { WorkflowStep } from './WorkflowStep';
```

Comment:
Should we be exporting any of the other exports in `WorkflowStep.ts`. Just wondering if some of those interfaces or types would be useful for typescript devs?

Reply:
Really happy you brought this up. Will wait on @aoberoi to chime in here.

Reply:
To come up with a recommendation, I'm looking for types that describe values which users might want to declare away from the actual call site. These are the situations where having a name for the type is useful.

For example, I think we should export `WorkflowStepOptions` because it's reasonable to want to write code like this:

```typescript
const stepOptions: WorkflowStepOptions = {
  edit: /* ... */,
  save: /* ... */,
  execute: /* ... */,
};

// some logic to manipulate the value imperatively
// when in debug mode, prepend some debugging middleware to each listener
if (debugMode) {
  for (const [interaction, listener] of Object.entries(stepOptions)) {
    const arrayOfListeners = Array.isArray(listener) ? listener : [listener];
    arrayOfListeners.unshift(debuggingMiddleware);
    stepOptions[interaction] = arrayOfListeners;
  }
}

// somewhere else
const step = new WorkflowStep('cid', stepOptions);
```

But we can't always predict which types might be useful to a developer ahead of time. It's tempting to just export all the types - someone might have a use for them, right? Except the more types we export, the more potential breakage we create when we update, rename, or remove those types. Remember: we don't use Semver to describe the TypeScript interface, only the JavaScript interface (we can talk about this more if anyone has questions). In my opinion, its better to err on the side of exporting fewer types now, to minimize the potential impact of type related API changes in the future.

All of this is to say, my recommendation is to additionally export `WorkflowStepOptions`, `WorkflowStepEditMiddleware`, `WorkflowStepSaveMiddleware`, and `WorkflowStepExecuteMiddleware`.